pkg/fleetautoscalers/controller.go

```diff
@@ -60,11 +60,15 @@ import (
 )
 
 // fasThread is used for tracking each Fleet's autoscaling jobs
-//
-//nolint:govet // ignore fieldalignment, one per fleet autoscaler
 type fasThread struct {
-	generation int64
 	cancel     context.CancelFunc
+	state      map[string]any
+	generation int64
+}
+
+// close cancels the context and cleans up any resources
+func (ft *fasThread) close(_ context.Context) {
```

Comment:
Does it actually needs the context here in parameter ? As it closes the one from the fasThread.cancel ? (which is set within addFasThread(), or maybe we want to close the context received from the parameter and removing the cancel from the struct ?) 🤔